### PR TITLE
[git-webkit] Gate information URL on initial setup

### PR DESCRIPTION
**Pull request by JonWBedard**
#### d81349865eba4d2749849ecd710a72b2d136a3ad
<pre>
[git-webkit] Gate information URL on initial setup
<a href="https://bugs.webkit.org/show_bug.cgi?id=237325">https://bugs.webkit.org/show_bug.cgi?id=237325</a>
&lt;rdar://problem/89620175 &gt;

Reviewed by Aakash Jain.

* Tools/Scripts/libraries/webkitscmpy/setup.py: Bump version.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/__init__.py: Ditto.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/setup.py:
(Setup.main): Offer to open documentation URL if repository has not been setup.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/setup_unittest.py:
(TestSetup.test_github):

Canonical link: <a href="https://commits.webkit.org/247933@main">https://commits.webkit.org/247933@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@290661">https://svn.webkit.org/repository/webkit/trunk@290661</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

### CODE REVIEW

File: Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/setup_unittest.py [hunk 1]
```diff
@@ -111,7 +109,7 @@ def test_git(self):
     def test_github_checkout(self):
         self.maxDiff = None
         with OutputCapture(level=logging.INFO) as captured, mocks.remote.GitHub() as remote, \
-            MockTerminal.input('n', 'committer@webkit.org', 'n', 'Committer', 'n', 'overwrite', 'disabled', '1', 'y', 'y'), \
+            MockTerminal.input('n', 'n', 'committer@webkit.org', 'n', 'Committer', 'n', 'overwrite', 'disabled', '1', 'y', 'y'), \
```
Comment (aj062) [inline Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/setup_unittest.py:112]: is this intentional?

File: Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/setup_unittest.py [hunk 2]
```diff
@@ -89,7 +87,7 @@ def test_git(self):
         self.assertEqual(
             captured.stdout.getvalue(),
             'For detailed information about options this script is configuring see:\n'
```
Comment (aj062) [inline Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/setup_unittest.py]: This is slightly twisted to read. Might re-word something like: 
'For detailed information about the options configured by this script, please see:\n'

File: Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/setup_unittest.py [hunk 3]
```diff
@@ -54,8 +54,6 @@ def test_github(self):
 
         self.assertEqual(
             captured.stdout.getvalue(),
-            'For detailed information about options this script is configuring see:\n'
-            'https://github.com/WebKit/WebKit/wiki/Git-Config#Configuration-Options\n'
```
Comment (aj062) [inline Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/setup_unittest.py]: is this removal intentional?

File: Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/setup.py [hunk 4]
```diff
@@ -301,12 +301,28 @@ def parser(cls, parser, loggers=None):
 
     @classmethod
     def main(cls, args, repository, **kwargs):
-        print('For detailed information about options this script is configuring see:')
-        print('https://github.com/WebKit/WebKit/wiki/Git-Config#Configuration-Options')
-
         if isinstance(repository, local.Git):
+            if 'true' != repository.config().get('webkitscmpy.setup', ''):
+                info_url = 'https://github.com/WebKit/WebKit/wiki/Git-Config#Configuration-Options'
```
Comment (aj062) [inline Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/setup.py:306]: can consider saving this url as a variable.

File: Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/setup.py [hunk 5]
```diff
@@ -301,12 +301,28 @@ def parser(cls, parser, loggers=None):
...
+                print('For detailed information about options this script is configuring see:')
```
Comment (aj062) [inline Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/setup.py]: Ditto about rewording

File: Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/setup.py [hunk 6]
```diff
@@ -301,12 +301,28 @@ def parser(cls, parser, loggers=None):
...
+                print(info_url)
+                if not args.defaults and Terminal.choose("Would you like to open this URL in your browser?", default='Yes') == 'Yes':
+                    run(['open', info_url])
+                print('')
```
Comment (aj062) [inline Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/setup.py]: maybe use print('\n\n') instead of two print statements.

File: Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/setup.py [hunk 7]
```diff
@@ -301,12 +301,28 @@ def parser(cls, parser, loggers=None):
...
     @classmethod
     def main(cls, args, repository, **kwargs):
-        print('For detailed information about options this script is configuring see:')
-        print('https://github.com/WebKit/WebKit/wiki/Git-Config#Configuration-Options')
-
         if isinstance(repository, local.Git):
+            if 'true' != repository.config().get('webkitscmpy.setup', ''):
+                info_url = 'https://github.com/WebKit/WebKit/wiki/Git-Config#Configuration-Options'
+                print('For detailed information about options this script is configuring see:')
+                print(info_url)
```
Comment (aj062) [inline Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/setup.py]: can combine this print statement with above print statement

File: Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/setup.py [hunk 8]
```diff
@@ -301,12 +301,28 @@ def parser(cls, parser, loggers=None):
 
     @classmethod
     def main(cls, args, repository, **kwargs):
-        print('For detailed information about options this script is configuring see:')
-        print('https://github.com/WebKit/WebKit/wiki/Git-Config#Configuration-Options')
-
         if isinstance(repository, local.Git):
+            if 'true' != repository.config().get('webkitscmpy.setup', ''):
```
Comment (aj062) [inline Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/setup.py:305]: should it be called `webkitscmpy.setup` or something like `git-webkit.setup`?

**Review by aj062 — APPROVED** [review]
r+ with comments

File: Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/setup.py (hunk 8 above)
Comment (JonWBedard) [reply Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/setup.py:305]: I've been prefixing with `webkitscmpy` since that's the layer that actually owns these names. It's not our first git config option: https://github.com/WebKit/WebKit/blob/main/metadata/project_config

File: Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/setup_unittest.py (hunk 3 above)
Comment (JonWBedard) [reply Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/setup_unittest.py]: Yes. We moved this line so it's associated with a local repository, not all repositories (you can run `git-webkit setup` on a remote GitHub repository). I did this because this line doesn't make much sense in the context of a remote repository, where we really only need to setup a user-owned fork.

File: Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/setup_unittest.py (hunk 1 above)
Comment (JonWBedard) [reply Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/setup_unittest.py:112]: Yes! We need to provide an answer to `Would you like to open this URL in your browser?`

**Comment by JonWBedard** [comment]
Landed https://commits.webkit.org/247933@main (d81349865eba)!
